src/models.jl

```diff
@@ -1,10 +1,84 @@
 _maybe_fill(x, n) = x
 _maybe_fill(x::Number, n) = fill(x, n)
 
+function nth_nearest_neighbors(g, v, n::Int)  #ToDo: Add test for this.
+  if n >= 2
+    return setdiff(
+      neighborhood(g, v, n), union(neighborhood(g, v, n - 1), neighborhood(g, v, n - 2))
+    )
+  else
+    return neighborhood(g, v, 1)
+  end
```

Comment:
```suggestion
  isone(n) && return neighbors(g, v)
  return setdiff(
    neighborhood(g, v, n), union(neighborhood(g, v, n - 1), neighborhood(g, v, n - 2))
  )
```
I prefer that style, where special cases are handled first followed by the general implementation, I think it is easier to read compared to nesting things in if-statements.

Also, why is `union(neighborhood(g, v, n - 1), neighborhood(g, v, n - 2))` needed? Isn't `neighborhood(g, v, n - 2)` already included in `neighborhood(g, v, n - 1)`?

Reply:
from what I could tell `neighborhood(g,v,n)` gives you all vertices that can be visited with by traversing n (non-unique) edges, i.e. you can go back on a given edge. the vertices that can be visited via an odd number of edges is an orthogonal set to those that can be visited by an even number. `neighborhood(g, v, n - 1)` shouldn't be necessary here actually.